.github/workflows/master.yml

```diff
@@ -1,156 +1,238 @@
 name: master
 
 on:
-  push:
```

Comment:
I am sort of cautious about fixing this file because it's already very invasive and removed many things. @dongjoon-hyun, I removed this for now but let me know or feel free to restore/recover later.

Reply:
+1. I think we can have smoke tests after commits are pushed, instead of running all tests again.  We can have a followup : https://issues.apache.org/jira/browse/SPARK-32266

Reply:
What do you mean by `smoke tests` if you are not running all tests? PR Builder cannot be the same because it's just based on the master branch snapshot on the starting time. So, I don't agree with 'all tests again'.
>  instead of running all tests again. 

However, please proceed what you need. Everything can be done after merging initial PR.

BTW, please note that this PR can cause a significant delay if there is no enough resource given from ASF foundation.

Reply:
@dongjoon-hyun Sorry about using the word "smoke test" without details. I meant we can run linter/build/maven build jobs after commits are pushed, just like what it is now. 

> please note that this PR can cause a significant delay if there is no enough resource given from ASF foundation.

That's my concern here. If we run all the tests on push, then there might be too many queued jobs.

Reply:
@gengliangwang . That's the reason why @HyukjinKwon is working with ASF INFRA. I believe that we should achieve that.